src/core_modules/capture-core/components/CardList/CardListItem.component.js

```diff
@@ -227,15 +221,23 @@ const CardListItemIndex = ({
             </div>
 
             {
-                getCustomBottomElements &&
-                getCustomBottomElements({
-                    item,
-                    navigationButtonsState: deriveNavigationButtonState(enrollmentType),
-                    programName: currentSearchScopeName,
-                })
+                renderCustomCardActions
+                &&
+                <div className={classes.buttonMargin}>
+                    {
+                        renderCustomCardActions({
+                            item,
+                            // we pass the programName because we have the case that the scope of the search
+                            // can be different that the scopeId from the url
+                            // this can happen for example when you are registering through the relationships
+                            programName: currentSearchScopeName,
+                            enrollmentType,
```

Comment:
In my try to make this more generic I moved away from the `navigationButtonState` which was very search page oriented. Now I only pass the enrollment type and then the buttons on the search page will decide what is to be rendered.